RZAssemblage/Private/RZJoinAssemblage.m

```diff
@@ -0,0 +1,144 @@
+//
+//  RZJoinAssemblage.m
+//  RZAssemblage
+//
+//  Created by Brian King on 1/29/15.
+//  Copyright (c) 2015 Raizlabs. All rights reserved.
+//
+
+#import "RZJoinAssemblage.h"
+#import "NSIndexPath+RZAssemblage.h"
+#import "RZAssemblage+Private.h"
+#import "RZAssemblageDefines.h"
+
+@implementation RZJoinAssemblage
+
+- (instancetype)initWithAssemblages:(NSArray *)assemblages
+{
+    self = [super init];
+    for ( RZAssemblage *assemblage in assemblages ) {
+        [self addMonitorsForObject:assemblage];
+        RZRaize([assemblage isKindOfClass:[RZAssemblage class]], @"All objects in RZJoinAssemblage must be an RZAssemblage");
+    }
+    _assemblages = assemblages;
+    return self;
+}
+
+- (NSString *)description
+{
+    return [NSString stringWithFormat:@"<%@: %p join %@", self.class, self, self.assemblages];
+}
+
+#pragma mark - RZAssemblage
+
+- (NSUInteger)countOfElements
+{
+    NSUInteger count = 0;
+    for ( RZAssemblage *assemblage in self.assemblages ) {
+        count += [assemblage countOfElements];
+    }
+    return count;
```

Comment:
Could just be `return [self.assemblages valueForKeyPath:@"@sum.countOfElements"]`